iib/web/models.py

```diff
@@ -309,8 +340,21 @@ def _state_to_json(state):
     @classmethod
     def from_json(cls, kwargs):
         # Validate all required parameters are present
-        required_params = {'bundles', 'binary_image'}
+        required_params = {'bundles', 'binary_image', 'operators'}
         optional_params = {'from_index', 'add_arches', 'cnr_token'}
+
+        if 'bundles' in kwargs:
```

Comment:
What if I specify `operators` in my JSON payload for the `add` endpoint? IIB will fail in unexpected ways. My preference is to have two methods `from_add_json` and `from_remove_json` that both wrap `from_json`. `from_json` could accept an additional keyword argument such as `additional_required_params=None`.  `from_json` should also be renamed to `_from_json` since it shouldn't be called directly anymore.

Reply:
I agree, yours is a more cleaner solution. I'll change it. Thanks!

Reply:
Though, it wouldn't fail in unexpected ways if you specify `operators` in the JSON payload for the `add` endpoint. It'll fail with Invalid params validation error.

Reply:
The solution you suggested will probably duplicate some of the code but I guess it's worth it for readability.